### DISCUSSION TITLE
Argo server did not recover from a connection drop from archive persistence layer (PostgreSQL)

### DISCUSSION THREAD

**Discussion by kaancfidan** [argoproj/argo-workflows#7300] [Q&A] [answered]
I initially wanted to create an issue for this but I'm not sure how to reproduce it.

We have been using Argo Workflows in our AKS cluster on Azure for a year now and recently we have added the workflow archive by connecting the Argo server to a managed Azure Database for PostgreSQL. It had been running without any problems for ~2 weeks and this morning we have seen that the workflow APIs were returning Internal Server Error status code and when we checked the logs this was repeating:

> time="2021-11-30T07:29:35.369Z" level=error msg="finished unary call with code Unknown" error="write tcp 10.244.33.119:35646->10.10.0.7:5432: write: connection reset by peer" grpc.code=Unknown grpc.method=ListWorkflows grpc.service=workflow.WorkflowService grpc.start_time="2021-11-30T07:29:35Z" grpc.time_ms=20.129 span.kind=server system=grpc

Basically it was not able to reach the PostgreSQL instance.

PostgreSQL server in question was not down. The same instance is also used by other parts of our system and is accessed from the same cluster. So whatever network problem happened I assume it must have been transient. I restarted the argo-server deployment and the problem was fixed. Another weird thing is that I do not see any failed connection statistics on PostgreSQL side.

Connection drops are expected but I don't understand why Argo did not recover from it. Any ideas why this could occur?

**Comment by kaancfidan** [comment]
I suspect it might be related to https://github.com/upper/db/issues/226.

**Comment by terrytangyuan** [comment, marked as answer]
Please only use either issue or discussion instead of duplicating in two places. This is a duplicate of https://github.com/argoproj/argo-workflows/issues/7442. Please continue the discussion over there.

**Reply by kaancfidan** [reply]
Thanks. I duplicated it as an issue as it started happening periodically to us and I wasn't getting anything of this discussion.

**Reply by terrytangyuan** [reply]
I see. Let’s see if we see any upvotes in the existing issue. Currently it doesn’t seem like anyone else is having the same issue.
